source/core/src/test/com/csse3200/game/inventory/items/ConsumableItemTest.java

```diff
@@ -0,0 +1,57 @@
+package com.csse3200.game.inventory.items;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.csse3200.game.extensions.GameExtension;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(GameExtension.class)
+public class ConsumableItemTest {
+    private static TestableItem1 item1;
+    private static ItemUsageContext context;
+
+    /**
+     * Concrete subclass of ConsumableItems for testing purposes.
+     */
+    private static class TestableItem1 extends ConsumableItem {
+        public int numUsed = 0;
+
+        public TestableItem1(String name, int itemCode, int limit, int quantity) {
+            super(name, quantity, quantity, isEmpty);
+        }
+
+        @Override
+        public void useItem(ItemUsageContext context) {
+            // For testing purposes only
+            numUsed++;
+            quantity--;
+        }
+
+        @BeforeEach
+        void setUp() { // Initialize TestableItem and ItemUsageContext
+            item1 = new TestableItem1("Test1", 1, 10, 1);
+            context = new ItemUsageContext();
+        }
+
+        @Test
+        void testIfNotEmpty() {
```

Comment:
Create a test which checks the consumption use - ie don't bother overriding the ConsumableItem useItem function (this is what we are trying to test). Create a test object with e.g. 3 quantity, and useItem 3 times and check that quantity is the correct value each time. Then run useItem one more time and assert an error is thrown.

Reply:
Implemented this into newest commit